src/lib.rs

```diff
@@ -7,23 +7,26 @@
 //!
 //! # Ownership and Lifetimes
 //!
-//! In Python, all objects are implicitly reference counted.
-//! In rust, we will use the `PyObject` type to represent a reference to a Python object.
-//!
-//! Because all Python objects potentially have multiple owners, the
-//! concept of Rust mutability does not apply to Python objects.
-//! As a result, this API will allow mutating Python objects even if they are not stored
-//! in a mutable Rust variable.
-//!
-//! The Python interpreter uses a global interpreter lock (GIL)
-//! to ensure thread-safety.
-//! This API uses a zero-sized `struct Python<'p>` as a token to indicate
-//! that a function can assume that the GIL is held.
-//!
-//! You obtain a `Python` instance by acquiring the GIL,
-//! and have to pass it into all operations that call into the Python runtime.
+//! Because all Python objects potentially have multiple owners, the concept of
+//! Rust mutability does not apply to Python objects.  As a result, PyO3 allows
+//! mutating Python objects even if they are not stored in a mutable Rust
+//! variable.
+//!
+//! In Python, all objects are implicitly reference counted.  The Python
+//! interpreter uses a global interpreter lock (GIL) to ensure thread-safety.
+//! This API uses a zero-sized `struct Python<'py>` as a token to indicate that
```

Comment:
```suggestion
//! Thus, we use `struct Python<'py>` as a token to indicate that
```
The old `This API` refers to `PyObject` but the sentence it occurs is removed now.

Reply:
Indeed, I glued this in too eagerly :)